src/handlers/actions/payment.ts

```diff
@@ -0,0 +1,65 @@
+import { utils } from 'ethers';
```

Comment:
This handler is a bit messy, and I will clean it up when we start handling more actions and I can identify some refactoring opportunities